app/models/tag.rb

```diff
@@ -1023,7 +1033,7 @@ def remove_sub_filters(sub_tag)
   before_update :check_canonical
   def check_canonical
     if self.canonical_changed? && !self.canonical?
-      self.async(:remove_canonical_associations)
+      remove_canonical_associations
```

Comment:
Also, I made a mistake -- it's not just about the tests. Switching the definition of `Tag.async` causes some tests to fail, but I realize now that's actually because it's revealing more bugs in the existing code. (Stemming from the question of when the job is enqueued vs. when the required information for that job is written to the database.) I can't just change `add_merger_associations` back to async and fix up the tests.

There are a couple of more complex ways to call it asynchronously, if that's what you think is best, but it would be a more extensive change than just reverting to `async(:add_merger_associations)`, and it might take a while to iron out the bugs.